packages/contracts/scripts/managing-dao-setup.ts

```diff
@@ -0,0 +1,379 @@
+  const psp = PluginSetupProcessor__factory.connect(
+    PLUGIN_SETUP_PROCESSOR_ADDRESS,
+    deployer
+  );
+
+  const pluginSetupRef: PluginSetupRefStruct = {
+    pluginSetupRepo: GOVERNANCE_PLUGIN_REPO_ADDRESS!,
+    versionTag: GovernancePluginsSetupParams.VERSION,
+  };
+
+  const tx = await psp.prepareInstallation(MANAGING_DAO_ADDRESS!, {
+    pluginSetupRef,
+    data: installData,
+  });
+  await tx.wait();
+
+  const preparedEvent = await findEvent<InstallationPreparedEvent>(
+    tx,
+    'InstallationPrepared'
+  );
+  if (!preparedEvent) {
+    throw new Error('Failed to get InstallationPrepared event');
+  }
+  console.log(
+    '- Deployed a MainVotingPlugin plugin at',
+    preparedEvent.args.plugin
+  );
+  console.log(
+    '- Deployed a MemberAccessPlugin plugin at',
+    preparedEvent.args.preparedSetupData.helpers[0]
+  );
+
+  return {
+    pluginAddress: preparedEvent.args.plugin,
+    helpers: preparedEvent.args.preparedSetupData.helpers,
+    permissions: preparedEvent.args.preparedSetupData.permissions,
+  };
+}
+
+async function applyInstallation(
+  preparedInstallation: Awaited<ReturnType<typeof prepareInstallation>>
+) {
+  const {helpers, permissions, pluginAddress} = preparedInstallation;
+  const mgmtDAO = DAO__factory.connect(MANAGING_DAO_ADDRESS!, deployer);
+
+  // Encode a call to execute() > applyInstallation
+  console.log('Installing the plugin to the managing DAO');
+  const applyInstallationParams: PluginSetupProcessor.ApplyInstallationParamsStruct =
+    {
+      plugin: pluginAddress,
+      helpersHash: hashHelpers(helpers),
+      permissions,
+      pluginSetupRef: {
+        pluginSetupRepo: GOVERNANCE_PLUGIN_REPO_ADDRESS!,
+        versionTag: GovernancePluginsSetupParams.VERSION,
+      },
+    };
+
+  console.log('- Allowing the PSP to manage permissions');
+  console.log('- Telling the PSP to apply the installation');
+  console.log("- Revoking the PSP's permission to manage permissions");
+
+  const actions: IDAO.ActionStruct[] = [
+    // Allow the PSP to install a plugin
+    {
+      to: MANAGING_DAO_ADDRESS!,
+      value: 0,
+      data: DAO__factory.createInterface().encodeFunctionData('grant', [
+        MANAGING_DAO_ADDRESS!,
+        PLUGIN_SETUP_PROCESSOR_ADDRESS!,
+        await mgmtDAO.ROOT_PERMISSION_ID(),
+      ]),
+    },
+    // Install the plugin
+    {
+      to: PLUGIN_SETUP_PROCESSOR_ADDRESS!,
+      value: 0,
+      data: PluginSetupProcessor__factory.createInterface().encodeFunctionData(
+        'applyInstallation',
+        [MANAGING_DAO_ADDRESS!, applyInstallationParams]
+      ),
+    },
+    // Revoke the PSP permission
+    {
+      to: MANAGING_DAO_ADDRESS!,
+      value: 0,
+      data: DAO__factory.createInterface().encodeFunctionData('revoke', [
+        MANAGING_DAO_ADDRESS!,
+        PLUGIN_SETUP_PROCESSOR_ADDRESS!,
+        await mgmtDAO.ROOT_PERMISSION_ID(),
+      ]),
+    },
+  ];
+
```

Comment:
Shouldn't we revoking ROOT_PERMISSION from the deployer to the managing dao as well ?

Reply:
Same

Reply:
This is done in the dropExecutePermission function below